header-validator/src/maybe.ts

```diff
@@ -1,29 +1,30 @@
-export type Maybe<T> = typeof None | Some<T>
+export type Maybeable<T> = T | Maybe<T>
 
-export const None = {
-  filter: () => None,
-  map: () => None,
-  flatMap: () => None,
-  peek: () => None,
-}
+export class Maybe<T> {
+  static readonly None: Maybe<any> = new Maybe()
 
-export class Some<T> {
-  constructor(private readonly t: T) {}
+  static some<T>(t: T): Maybe<T> {
+    return new Maybe(t)
+  }
 
-  filter(f: (t: T) => boolean): Maybe<T> {
-    return f(this.t) ? this : None
+  static flatten<T>(t: Maybeable<T>): Maybe<T> {
```

Comment:
Could it be private ?

Reply:
It will be used in the followup PR, but noted.